SignalMessaging/Views/ThreadViewHelper.m

```diff
@@ -48,49 +52,117 @@ - (void)initializeMapping
         [[YapDatabaseViewMappings alloc] initWithGroups:@[ grouping ] view:TSThreadDatabaseViewExtensionName];
     [self.threadMappings setIsReversed:YES forGroup:grouping];
 
-    __weak ThreadViewHelper *weakSelf = self;
-    [self.uiDatabaseConnection asyncReadWithBlock:^(YapDatabaseReadTransaction *transaction) {
-        [self.threadMappings updateWithTransaction:transaction];
+    self.uiDatabaseConnection = [TSStorageManager.sharedManager newDatabaseConnection];
+    [self.uiDatabaseConnection beginLongLivedReadTransaction];
 
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [weakSelf updateThreads];
-            [weakSelf.delegate threadListDidChange];
-        });
-    }];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(applicationWillEnterForeground:)
+                                                 name:UIApplicationWillEnterForegroundNotification
+                                               object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self
```

Comment:
Its expensive to rebuild our mapping every time we get an "external database modification" event, so stop observing DB changes in the main app when it is in the background and rebuild our mapping whenever we re-enter the foreground.

Reply:
Note that home and conversation view already do this.